src/Traits/FieldAssertions.php

```diff
@@ -24,13 +29,63 @@ trait FieldAssertions
      */
     public function assertHasField(string $field, string $message = ''): self
     {
-        PHPUnit::assertThat(
-            $this->component->fields(NovaRequest::createFromGlobals()),
-            new HasField($field, $this->allowPanels()),
-            $message
-        );
+        return $this->assertHasFieldIn('fields', NovaRequest::class, $field, $message);
+    }
 
-        return $this;
+    /**
+     * Checks that this Nova component has an index field with the same name or attribute.
+     *
+     * @param  string  $field  The name or attribute of the field
+     * @param  string  $message
+     * @return $this
+     */
+    public function assertHasIndexField(string $field, string $message = ''): self
+    {
+        $this->ensureIsResource();
+
+        return $this->assertHasFieldIn('fieldsForIndex', ResourceIndexRequest::class, $field, $message);
+    }
+
+    /**
+     * Checks that this Nova component has a detail field with the same name or attribute.
+     *
+     * @param  string  $field  The name or attribute of the field
+     * @param  string  $message
+     * @return $this
+     */
+    public function assertHasDetailField(string $field, string $message = ''): self
+    {
+        $this->ensureIsResource();
+
+        return $this->assertHasFieldIn('fieldsForDetail', ResourceDetailRequest::class, $field, $message);
+    }
+
+    /**
+     * Checks that this Nova component has a detail field with the same name or attribute.
+     *
+     * @param  string  $field  The name or attribute of the field
+     * @param  string  $message
+     * @return $this
+     */
+    public function assertHasCreationField(string $field, string $message = ''): self
+    {
+        $this->ensureIsResource();
+
+        return $this->assertHasFieldIn('fieldsForUpdate', ResourceCreateOrAttachRequest::class, $field, $message);
```

Comment:
@andypooletrioteca Not sure if I'm missing something here, but it looks like the `$fieldsMethod` parameters should be flipped: `assertHasCreationField` → `fieldsForCreate`